src/main/java/com/relayrides/pushy/apns/ApnsConnection.java

```diff
@@ -280,13 +281,17 @@ protected void channelRead0(final ChannelHandlerContext context, final RejectedN
 
 			// Regardless of the cause, we ALWAYS want to notify listeners that some sent notifications were not
 			// processed by the gateway (assuming there are some such notifications).
-			final Collection<T> unprocessedNotifications =
-					this.apnsConnection.sentNotificationBuffer.getAllNotificationsAfterSequenceNumber(
-							rejectedNotification.getSequenceNumber());
-
-			if (!unprocessedNotifications.isEmpty()) {
-				if (this.apnsConnection.listener != null) {
-					this.apnsConnection.listener.handleUnprocessedNotifications(this.apnsConnection, unprocessedNotifications);
+			// Unless the sequence number was erroneously reported as 0, in which case we don't know the actual
+			// sequence number, and can't determine what was sent after the bad notification.
+			if (rejectedNotification.getSequenceNumber() != 0) {
```

Comment:
Actually, one more thought—do we want to check the rejection reason here, too? Wrapping to `0` is, as we've all already acknowledged, really unlikely, but checking the reason might make it clearer what's going on (i.e. that we're working around a weird upstream issue).